client/src/rr/dnssec/keypair.rs

```diff
@@ -241,11 +242,12 @@ impl KeyPair {
     /// # Return
     ///
     /// the KEY record data
+    #[allow(deprecated)]
     pub fn to_sig0key(&self, algorithm: Algorithm) -> DnsSecResult<KEY> {
         self.to_public_bytes()
             .map(|bytes| {
                      KEY::new(Default::default(),
-                              Default::default(),
+                              KeyUsage::Zone,
```

Comment:
> Well that's only a default value for dnssec-keygen (and, according to the manpage, only for DNSKEY records).

Looking at the dnssec-keygen sources, I think given `-T KEY`, there is no default for the nametype:

From dnssec-keygen.c, at around line 843 (note that the `-T` argument sets the `options |= DST_TYPE_KEY;` flag):
```
	if (nametype == NULL) {
		if ((options & DST_TYPE_KEY) != 0) /* KEY / HMAC */
			fatal("no nametype specified");
		flags |= DNS_KEYOWNER_ZONE;	/* DNSKEY */
	} else if (strcasecmp(nametype, "zone") == 0)
		flags |= DNS_KEYOWNER_ZONE;
	else if ((options & DST_TYPE_KEY) != 0)	{ /* KEY / HMAC */
		if (strcasecmp(nametype, "host") == 0 ||
			 strcasecmp(nametype, "entity") == 0)
			flags |= DNS_KEYOWNER_ENTITY;
		else if (strcasecmp(nametype, "user") == 0)
			flags |= DNS_KEYOWNER_USER;
		else
			fatal("invalid KEY nametype %s", nametype);
	} else if (strcasecmp(nametype, "other") != 0) /* DNSKEY */
		fatal("invalid DNSKEY nametype %s", nametype);
```

> You can change it with -n HOST or -n USER.

Good point. I certainly used the arguments `-T KEY` and `-n ZONE` to generate the keys that worked for me (with nsupdate) and to which I compared later on. But I tried neither `-n entity` nor `-n user`.

Maybe require a specific `KeyUsage` without providing a default, just like Bind's `dnssec-keygen` does?